autonat.go

```diff
@@ -187,20 +204,55 @@ func (as *AmbientAutoNAT) background() {
 
 		case e := <-subChan:
 			switch e := e.(type) {
-			case event.EvtLocalAddressesUpdated:
-				if !lastAddrUpdated.Add(time.Second).After(time.Now()) {
-					lastAddrUpdated = time.Now()
-					if as.confidence > 1 {
-						as.confidence--
-					}
-				}
 			case event.EvtPeerIdentificationCompleted:
 				if s, err := as.host.Peerstore().SupportsProtocols(e.Peer, AutoNATProto); err == nil && len(s) > 0 {
-					currentStatus := as.status.Load().(autoNATResult)
-					if currentStatus.Reachability == network.ReachabilityUnknown {
+					currentStatus := as.status.Load().(natStatus)
+					if currentStatus.reachability == network.ReachabilityUnknown {
 						as.tryProbe(e.Peer)
 					}
 				}
+
+			case event.EvtDiscoveredAddresses:
+				switch e.Source {
```

Comment:
I think these are the right categories (approximately), but I'd make them more about the category/confidence and less about the source:

1. Configured (user config, maybe that's not even an event) - the user told me.
2. Interface (listen address) - my computer told me.
3. Network(?) (NAT, gateway, docker, aws, etc.). - a computer on the network told me.
4. Reported (observation). - a random computer on the internet told me.

It doesn't really matter that the observation came in from identify, it's an observation. It doesn't really matter if the event was from the NAT port mapping system, or some docker probe, etc.